lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java

```diff
@@ -409,4 +477,46 @@ private ImageAssetBitmapManager getImageAssetBitmapManager() {
     }
     callback.unscheduleDrawable(this, what);
   }
+
+  private static class ColorFilterData {
+
+    final String layerName;
+    @Nullable final String contentName;
+    @Nullable final ColorFilter colorFilter;
+    private final int hashCode;
+
+    ColorFilterData(String layerName, @Nullable String contentName,
+        @Nullable ColorFilter colorFilter) {
+      this.layerName = layerName;
+      this.contentName = contentName;
+      this.colorFilter = colorFilter;
+      hashCode = (this.layerName + this.contentName).hashCode();
```

Comment:
3 questions here:
1) Do you need equals and hashCode for this?
2) Why are you computing this in the constructor?
3) If you do need this, you can just move it to the hashCode function but the proper way to do this would be:
```
int hashCode = 17;
hashCode = hashCode * 31 * layerName.hashCode();
hashCode = hashCode * 31 * contentName.hashCode();
```
Then you don't have to do a string allocation either.